### PR TITLE
Nahiyan_Optimized dashboard for mobile view

### PR DESCRIPTION
**Pull request by Nahiyan-16**
# Description
ALL THE RED LINES FIXED
Screenshot 2024-08-24 130520

## Related PRS (if any):
This frontend PR is related to the dev backend

## Main changes explained:
- Aligned dashboard components when 991px and below
- Created new dropdown for days sector when below 768px.
- Aligned Tracking/Hide button to the right

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as owner user
5. go to dashboard
6. verify dashboard components are aligned when 991px and below
7. verify that days selector displays correctly when 768px and below
8. verify that Tracking/Hide button is aligned to the right.
9. verify these new features works in [dark mode](https://docs.google.com/document/d/11OXJfBBedK6vV-XvqWR8A9lOH0BsfnaHx01h1NZZXfI)

## Screenshots or videos of changes:
Screenshot 2024-08-24 115005
Screenshot 2024-08-24 130151

### CODE REVIEW

**Review by Parth-tech — DISMISSED** [review]
Hey Nahiyan,
The UI works well at varying screen sizes and in both light and dark mode.
Great job done.

`Image references:`

Screenshot 2024-08-24 at 10 47 10 AM
Screenshot 2024-08-24 at 10 42 16 AM

**Review by Pranay-ai — APPROVED** [review]
Hey @Nahiyan-16 
The PR is Working as expected please find the related screenshots.
image
image
image
Thank You.

**Review by kdaga81 — CHANGES_REQUESTED** [review]
image

The date selector's dropdown is displayed in the middle. Please check the image for reference.

**Review by SnelPatare — APPROVED** [review]
Working as expected!!

image

image

**Review by manikittu810 — DISMISSED** [review]
Hello @Nahiyan-16 
The changes are working as mentioned 
Screenshot 2024-08-24 at 3 25 56 PM
Thank you.

**Review by oliviahyw — CHANGES_REQUESTED** [review]
Hi @Nahiyan-16 I have tested this PR. The date selector is not working as expected. Please check the screenshot attached.
PR2614

**Review by vishnupriyaatheti — APPROVED** [review]
@Nahiyan-16 I have reviewed your PR as per the given 5 points  in redlines. All changes are working fine and below are my observations:
1) Badges, weekly summary.....etc seems to be in same width at given px.
ii) Weekly summary checkbox is getting displayed at given 544px
iii) days icon: Displaying as expected.
iv) Tracking button check: till 768px it shows at fixed position right, Above 768px it is floats to the middle.
v) from 345px it the title "View time .." going to second line. At 375px it looks as expected.
iv) Search button is working as expected.
FE PR 2614(i) above 991px check
FE PR 2614(i) check
FE PR 2614(ii) weekly checkbox check below 544px
FE PR 2614(iii) below 480px check
FE PR 2614(iv) above 544px at 768px
FE PR 2614(iv) above 544px tracking button check
FE PR 2614(iv) below 544px tracking button check
FE PR 2614(v) at 345px search button check
FE PR 2614(v) at 375px search button check

**Comment by FayeLV** [comment]
Hello @Nahiyan-16
The changes are working as shown in the screenshots.

Snipaste_2024-08-27_15-34-34
Snipaste_2024-08-27_15-35-50
Snipaste_2024-08-27_15-36-58

**Review by preethamdnr — APPROVED** [review]
Hello Nahiyan,
This PR is working as expected in both the modes with varying dimensions.
image
image

**Review by FayeLV — APPROVED** [review]

**Review by yashwanth170 — APPROVED** [review]
Hey @Nahiyan-16
The PR is Working as expected. I have tested responsiveness for various dimensions and its working as expected. 
Days

**Review by JatinAgrawal94 — DISMISSED** [review]
Changes work as expected. Good work.
Screenshot 2024-08-28 at 8 54 05 PM
Screenshot 2024-08-28 at 8 54 49 PM

File: src/components/TeamMemberTasks/TeamMemberTasks.jsx [hunk 1]
```diff
@@ -599,18 +598,48 @@ const TeamMemberTasks = React.memo(props => {
                   }}
                   onClick={() => selectPeriod(days)}
                 >
-                  {days} days
+                  {days} {days === "1" ? "day" : "days"}
```
Comment (crystallow1168) [inline src/components/TeamMemberTasks/TeamMemberTasks.jsx:601]: What do you think about DRYing up this code block? We seem to use this line more than once (line 601 and 630).

**Comment by crystallow1168** [comment]
Hello @Nahiyan-16,
I have reviewed your PR, I noticed a couple of things:

- ❗️ Dashboard components are not aligned when screen is 991px and below


https://github.com/user-attachments/assets/0823111f-ff9e-4238-8b3a-bd82f0d612d3



- ✅ Dropdown for days sector is displayed when screen is below 768px. 
  - ❗️The dropdown day selector seems to float in the middle. 
image


- ✅ Tracking/Hide button is aligned to the right.

https://github.com/user-attachments/assets/1245fc88-8519-4b34-a033-d7853ff9cc7b

**Review by crystallow1168 — CHANGES_REQUESTED** [review]
Hello @Nahiyan-16,
I have reviewed your PR, I noticed a couple of things:

- ❗️ Dashboard components are not aligned when screen is 991px and below


https://github.com/user-attachments/assets/0823111f-ff9e-4238-8b3a-bd82f0d612d3



- ✅ Dropdown for days sector is displayed when screen is below 768px. 
  - ❗️The dropdown day selector seems to float in the middle. 
image


- ✅ Tracking/Hide button is aligned to the right.

https://github.com/user-attachments/assets/1245fc88-8519-4b34-a033-d7853ff9cc7b

**Review by gowthamdongari — APPROVED** [review]
Hello,
I have reviewed this PR and working as expected.
image

**Review by vishavk1992 — DISMISSED** [review]
I have tested this PR and it works as expected for mobile view.

Screenshot 2024-08-30 162937

**Review by vedantg24 — APPROVED** [review]
Tested the PR, and the changes are working as expected. At 375px, the title "Viewing time entries from..” looks as expected, but below that, it goes to the second line
Screenshot 2024-09-03
Screenshot 2024-09-03
Screenshot 2024-09-03
Screenshot 2024-09-03
Screenshot 2024-09-03

**Review by AJAYINAVOLU — APPROVED** [review]
Changes are working as expected.

image

Thank you

**Review by Sudershhh — COMMENTED** [review]
The functionalities work as expected however thre might be one minor pixel issue with the days alignment. the days dropdown only seem to work below 767px. Let me know if you can recreate this.

**Pull request merged**